heartbeat/docs/reference/configuration/heartbeat-options.asciidoc

```diff
@@ -0,0 +1,448 @@
+access.
+* `tcp`: Connects via TCP and optionally verifies the endpoint by sending and/or
+receiving a custom payload.
+* `http`: Connects via HTTP and optionally verifies that the host returns the
+expected response.
+
+[[monitor-name]]
+===== name
+
+The monitor name. This value appears in the <<exported-fields,exported fields>>
+under the `monitor` field as the job name and the `type` field as the job type.
+
+[[monitor-enabled]]
+===== enabled
+
+A Boolean value that specifies whether the module is enabled. If the `enabled`
+option is missing from the configuration block, the module is enabled by
+default.
+
+[[monitor-schedule]]
+===== schedule
+
+A cron-like expression that specifies the task schedule. For example:
+
+* `*/5 * * * * * *` runs the task every 5 seconds (for example, at 10:00:00,
+10:00:05, and so on).
+* `@every 5s` runs the task every 5 seconds from the time when Heartbeat was
+started.
+
+//Steffen: Should we document the syntax of the cron expression somewhere, or point to a resource if one is available that describes specifically what the scheduler supports
+
+[[monitor-ipv4]]
+===== ipv4
+
+A Boolean value that specifies whether to ping using the ipv4 protocol if
+hostnames are configured. The default is `true`.
+
+[[monitor-ipv6]]
+===== ipv6
+
+A Boolean value that specifies whether to ping using the ipv6 protocol
+if hostnames are configured. The default is `true`.
+
+[[monitor-mode]]
+===== mode
+
+If `mode` is `any`, the monitor pings only one IP address for a hostname. If
+`mode` is `all`, the monitor pings all resolvable IPs for a hostname. The
+`mode: all` setting is useful if you are using a DNS-load balancer and want to
+ping every IP address for the specified hostname. The default is `any`.
+
+[[monitor-timieout]]
+===== timeout
+
+The total running time for each ping test. This is the total time allowed for
+testing the connection and exchanging data. The default is 16 seconds (16s).
+
+//Steffen: What happens if this timeout is exceeded?
+
+[[monitor-watch-poll-file]]
+===== watch.poll_file
+
+experimental[]
+
+The JSON file to watch for additional monitor configurations. The JSON file can
+have multiple objects, each of which specifies a different monitor config.
+Heartbeat checks this file periodically and starts a new monitor for any new
+objects that are added to the file. For example, imagine that you add 10
+new entries to the JSON file, each for a different hostname. When Heartbeat
+picks up the changes in the file, it starts a monitor for each host that you've
+configured. In this way, the configuration that you specify for the monitor in
+the main Heartbeat config file acts like a default config that you can
+live-reconfigure by specifying additional configurations in the external JSON
+file. 
+
+//Steffen: Not sure I got that quite right, but there you go! BTW, if you delete an object from the JSON file, I guess it stops the monitor that it has running for that object, right?
```

Comment:
@urso So what if the JSON file contains a config for a host that you're already monitoring in the main config file? Do the settings in the JSON file override what you've specified in the Heartbeat config? In other words, can you use the JSON file to override settings in the main config file, or just to add configs? I'm assuming the latter (because you probably want to allow users to set up multiple monitors for the same host), but I'm not sure.